src/Compilers/CSharp/Portable/BoundTree/OutVarLocalPendingInference.cs

```diff
@@ -2,20 +2,19 @@
 
 using Microsoft.CodeAnalysis.CSharp.Symbols;
 using Microsoft.CodeAnalysis.CSharp.Syntax;
+using System.Diagnostics;
 
 namespace Microsoft.CodeAnalysis.CSharp
 {
     internal partial class OutVarLocalPendingInference
     {
         public BoundLocal SetInferredType(TypeSymbol type, bool success)
         {
-            var syntaxNode = (DeclarationExpressionSyntax)this.Syntax;
-            Binder.DeclareLocalVariable(
-                (SourceLocalSymbol)this.LocalSymbol,
-                syntaxNode.Identifier(),
-                type);
+            Debug.Assert((object)type != null);
+            Debug.Assert(this.Syntax is DeclarationExpressionSyntax);
```

Comment:
Yes, will change.

---
In reply to: [75729099](https://github.com/dotnet/roslyn/pull/13274#discussion_r75729099) [](ancestors = 75729099)